safe/gui/tools/multi_exposure_dialog.py

```diff
@@ -540,21 +523,43 @@ def accept(self):
             self._multi_exposure_if = MultiExposureImpactFunction()
             self._multi_exposure_if.hazard = hazard
             self._multi_exposure_if.exposures = exposures
-            self._multi_exposure_if.aggregation = aggregation
+            if aggregation:
+                self._multi_exposure_if.aggregation = aggregation
+            else:
+                self._multi_exposure_if.crs = (
+                    self.iface.mapCanvas().mapSettings().destinationCrs())
             self._multi_exposure_if.debug = False
             self._multi_exposure_if.callback = self.progress_callback
 
             status, message = self._multi_exposure_if.prepare()
-            # self.assertEqual(code, PREPARE_MULTI_SUCCESS, message)
             if status == ANALYSIS_FAILED_BAD_INPUT:
                 self.hide_busy()
                 LOGGER.info(
                     'The impact function could not run because of the inputs.')
                 send_error_message(self, message)
                 LOGGER.info(message.to_text())
-            else:
+
+            elif status == PREPARE_SUCCESS:
                 code, message = self._multi_exposure_if.run()
-                # self.assertEqual(code, ANALYSIS_MULTI_SUCCESS, message)
+                if status == ANALYSIS_FAILED_BAD_INPUT:
+                    self.hide_busy()
+                    LOGGER.info(tr(
+                        'The impact function could not run because of the '
```

Comment:
"because one or more of the input layers were not suitable for use." .. .I wonder if we can provide some tips on which layer(s) was / were bad and what the issue encountered was....

Reply:
Everything is in the `message` variable